packages/core/src/app/ovm/deciders/db-decider.ts

```diff
@@ -0,0 +1,70 @@
+import { Md5 } from 'ts-md5/dist/md5'
+
+import { Decider, Decision } from '../../../types/ovm/decider.interface'
+import { Bucket, DB } from '../../../types/db'
+
+export abstract class DbDecider implements Decider {
+  private readonly decisionBucket: Bucket
+
+  protected constructor(db: DB) {
+    this.decisionBucket = db.bucket(Buffer.from(this.getUniqueId()))
+  }
+
+  public async checkDecision(input: any): Promise<Decision> {
+    const hash: Buffer = DbDecider.getCacheKey(input)
+    const decisionBuffer: Buffer = await this.decisionBucket.get(hash)
+
+    if (decisionBuffer === null) {
+      return undefined
+    }
+
+    return this.deserializeDecision(decisionBuffer)
+  }
```

Comment:
Hmm does this only work for decisions on particular keys? For for instance, I don't think this works if we want a decision over a range.

Reply:
Yeah, this will likely need to be overridden at times. If the `Decision` is directly associated with the `input` this should work, but in the case where the `Decision` may be associated with _another input_ that is derived from the input, it will not work.